### DISCUSSION TITLE
Example where the paragraph is rendering with out referring to the layout is confusing

### DISCUSSION THREAD

**Discussion by Hakuz** [a-h/templ#736] [Q&A]
Please help me understand how this is working. How the paragraph is referring to the layout in the provided example in this page https://templ.guide/syntax-and-usage/template-composition

> Components as parameters
> Components can also be passed as parameters and rendered using the @component expression.
> 
> package main
> 
> templ heading() {
>     <h1>Heading</h1>
> }
> 
> templ layout(contents templ.Component) {
> 	<div id="heading">
> 		@heading()
> 	</div>
> 	<div id="contents">
> 		@contents
> 	</div>
> }
> 
> templ paragraph(contents string) {
> 	<p>{ contents }</p>
> }
> 
> main.go
> package main
> 
> import (
> 	"context"
> 	"os"
> )
> 
> func main() {
> 	c := paragraph("Dynamic contents")
> 	layout(c).Render(context.Background(), os.Stdout)
> }

**Comment by joerdav** [comment]
Hi, the answer is that the paragraph does not refer to the layout in anyway in this example. The composition happens in the `main` function.

`templ layout(contents templ.Component)` - the layout can take the function argument of a templ component.
`templ paragraph(contents string)` - paragraph is a simple component that just takes a string.

In the `main` function we create an instance of the paragraph component: 

```go
c := paragraph("Dynamic contents")
```

And then we pass this component into the layout, and render the result:

```go
layout(c).Render(context.Background(), os.Stdout)
```
